packages/query-core/src/types.ts

```diff
@@ -147,12 +147,13 @@ export interface QueryOptions<
    * Setting it to `Infinity` will disable garbage collection.
    */
   gcTime?: number
-  queryFn?: QueryFunction<TQueryFnData, TQueryKey, TPageParam>
+  queryFn?: QueryFunction<TQueryFnData, TQueryKey, TPageParam> | SkipToken
```

Comment:
We can either keep it in here or only implement the logic in react package.

I'm okay with both options.

If we implement this only in react package then other packages might implement it in very similar way and we won't add anything to query-core.